trace_decoder/tests/trace_decoder_tests.rs

```diff
@@ -0,0 +1,272 @@
+//! ```
+
+use std::time::Duration;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use alloy_rpc_types_eth::Header;
+use anyhow::Context as _;
+use evm_arithmetization::prover::testing::simulate_execution;
+use evm_arithmetization::GenerationInputs;
+use itertools::Itertools;
+use log::{info, warn};
+use mpt_trie::partial_trie::PartialTrie;
+use plonky2::field::goldilocks_field::GoldilocksField;
+use plonky2::util::timing::TimingTree;
+use plonky2_maybe_rayon::*;
+use pretty_env_logger::env_logger::{try_init_from_env, Env, DEFAULT_FILTER_ENV};
+use prover::BlockProverInput;
+use rstest::rstest;
+use trace_decoder::OtherBlockData;
+
+type F = GoldilocksField;
+
+const JERIGON_WITNESS_DIR: &str = "tests/data/witnesses/zero_jerigon";
+//const CDK_ERIGON_WITNESS_DIR: &str =
+// "tests/data/witnesses/hermez_cdk_erigon";
+
+enum JsonFileType {
+    Witness,
+}
+
+fn init_logger() {
+    let _ = try_init_from_env(Env::default().filter_or(DEFAULT_FILTER_ENV, "info"));
+}
+
+fn find_json_data_files(dir: &str, file_type: JsonFileType) -> anyhow::Result<Vec<PathBuf>> {
+    let read_dir = fs::read_dir(dir)?;
+    read_dir
+        .into_iter()
+        .map(|dir_entry| dir_entry.map(|it| it.path()))
+        .filter_ok(|path| match file_type {
+            JsonFileType::Witness => !path
+                .to_str()
+                .expect("valid str file path")
+                .contains("header"),
+        })
+        .collect::<Result<Vec<_>, _>>()
+        .context(format!("Failed to find witness files in dir {dir}"))
+}
+
+fn read_witness_file(file_path: &Path) -> anyhow::Result<Vec<BlockProverInput>> {
+    let witness = fs::File::open(file_path).context("Unable to read file")?;
+    let mut reader = std::io::BufReader::new(witness);
+    let jd = &mut serde_json::Deserializer::from_reader(&mut reader);
+    serde_path_to_error::deserialize(jd).context(format!(
+        "Failed to deserialize json file {}",
+        file_path.display()
+    ))
+}
+
+fn decode_generation_inputs(
+    block_prover_input: BlockProverInput,
+) -> anyhow::Result<Vec<GenerationInputs>> {
+    let block_num = block_prover_input.other_data.b_data.b_meta.block_number;
+    let trace_decoder_output = trace_decoder::entrypoint(
+        block_prover_input.block_trace,
+        block_prover_input.other_data.clone(),
+        |_| unimplemented!(),
+    )
+    .context(format!(
+        "Failed to execute trace decoder on block {}",
+        block_num
+    ))?
+    .into_iter()
+    .collect::<Vec<GenerationInputs>>();
+    Ok(trace_decoder_output)
+}
+
+fn verify_generation_inputs(
+    header: &Header,
+    other: &OtherBlockData,
+    generation_inputs: Vec<GenerationInputs>,
+) -> anyhow::Result<()> {
+    if generation_inputs.is_empty() {
+        warn!("Empty generation inputs list");
+        return Ok(());
+    }
```

Comment:
This cannot happen. You can assert that `generation_inputs.len() >= 2`.
We always need at least 2 payloads for aggregation on the zero-bin side to work.